### PR TITLE
chore: define examples router & add ellipsis comp

### PR DESCRIPTION
**Pull request by wangjue666**
添加组件路由

## Summary by CodeRabbit

- **New Features**
	- Added localization support for component titles in both English and Chinese, enhancing accessibility.
	- Introduced a new routing module for a component-based page in the application.
	- Added routes for component demos, improving application organization and user experience.
	- Established a new component for visually representing password strength, enhancing user experience in secure password input.
	- Introduced a tooltip component for customizable content display, improving user interaction.
	- Added an ellipsis text component to manage text overflow and enhance UI clarity.
	- Enhanced tooltip customization options with new style properties.
	- Added a new `longText` data constant to showcase project features.
	- Released a new component for managing text overflow with an optional tooltip, significantly improving user experience.

- **Improvements**
	- Enhanced the formatting and organization of localization JSON files for better readability.

### CODE REVIEW

**Review by likui628 — APPROVED** [review]

File: apps/web-antd/src/router/routes/modules/demos.ts [hunk 1]
```diff
@@ -167,6 +167,16 @@ const routes: RouteRecordRaw[] = [
...
       },
+      // 组件
+      {
+        meta: {
+          icon: 'ion:layers-outline',
+          title: $t('page.demos.comp.title'),
+        },
+        name: 'CompDemos',
+        path: '/demos/comp',
+        children: [],
```
Comment (likui628) [inline apps/web-antd/src/router/routes/modules/demos.ts:178]: children为空，且component为空会报错

Comment (wangjue666) [reply apps/web-antd/src/router/routes/modules/demos.ts:178]: 是呢，先设置为空，要迁移的v2版本的组件，写好组件页面往这里加

**Comment by anncwb** [comment]
加个菜单：示例(examples)，跟演示同级。把这些组件相关的都放里面去应该好点。演示下面后面菜单层级应该会挺多

**Review by likui628 — DISMISSED** [review]

**Review by vince292007 — APPROVED** [review]

File: apps/web-antd/src/router/routes/modules/demos.ts (hunk 1 above)
Comment (vince292007) [reply apps/web-antd/src/router/routes/modules/demos.ts]: 目前这里确定不影响是吗

**Comment by vince292007** [comment]
@wangjue666 等你加个具体的菜单在一起合并，目前合并会有空菜单

**Comment by wangjue666** [comment]
> @wangjue666 等你加个具体的菜单在一起合并，目前合并会有空菜单

好的

**Review by likui628 — APPROVED** [review]

**Comment by wangjue666** [comment]
cc @vince292007  @anncwb  看看有需要调整的地方吗

File: packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue [hunk 2]
```diff
@@ -0,0 +1,119 @@
+<script setup lang="ts">
+import { computed, nextTick, ref, watchEffect } from 'vue';
+import type { CSSProperties } from 'vue';
+
+import { VbenTooltip } from '@vben-core/shadcn-ui/components/tooltip';
```
Comment (anncwb) [inline packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue:5]: 这里 @vben-core/shadcn-ui 即可

File: packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+</template>
+
+<style>
+.pointer {
+  @apply cursor-pointer;
+}
+</style>
+
+<style module>
+.ellipsisOneLine {
```
Comment (anncwb) [inline packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue:111]: 应该可以直接用tailwindcss在上面写

File: packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  maxWidth: '100%',
+  placement: 'top',
+  tooltip: true,
+  tooltipBackgroundColor: 'red',
+  tooltipColor: '',
+  tooltipFontSize: 14,
+  tooltipMaxWidth: undefined,
+  tooltipOverlayStyle: () => ({ textAlign: 'justify' }),
+});
+const emit = defineEmits(['expandChange']);
```
Comment (anncwb) [inline packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue:31]: defineEmits<{expandChange:[boolean]}>()

File: packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue [hunk 5]
```diff
@@ -0,0 +1,119 @@
+<script setup lang="ts">
+import { computed, nextTick, ref, watchEffect } from 'vue';
+import type { CSSProperties } from 'vue';
+
+import { VbenTooltip } from '@vben-core/shadcn-ui/components/tooltip';
+
+interface Props {
+  expand?: boolean; // 是否启用点击文本展开全部
```
Comment (anncwb) [inline packages/effects/common-ui/src/ellipsis-text/ellipsis-text.vue:8]: props 可以调整下注释格式 ，有默认值的可以简单写下
/** 
* 是否启用点击文本展开全部
* @default false
* */

**Review by anncwb — CHANGES_REQUESTED** [review]

**Review by vince292007 — APPROVED** [review]

**Pull request merged**